primefaces/src/main/java/org/primefaces/util/FileUploadUtils.java

```diff
@@ -224,111 +222,105 @@ protected static String convertJavaScriptRegex(String jsRegex) {
                     }
-                    break;
-                }
-                //Now we have a media type that may contain wildcards
-                if (FilenameUtils.wildcardMatch(contentType.toLowerCase(), accept)) {
-                    accepted = true;
-                    if (LOGGER.isLoggable(Level.FINE)) {
-                        LOGGER.fine(String.format("The content type %s of the uploaded file %s is accepted by %s", contentType, fileName, accept));
+                    // or try with IANA media types
+                    if (FilenameUtils.wildcardMatch(contentTypeLC, accept)) {
+                        LOGGER.log(Level.FINE,
+                                () -> String.format("Content type %s of the uploaded file %s is accepted by %s", contentTypeLC, fileName, accept));
+                        return true;
                     }
-                    break;
-                }
-            }
-            if (!accepted) {
-                if (LOGGER.isLoggable(Level.FINE)) {
-                    LOGGER.fine(String.format("The uploaded file %s with content type %s does not match the accept specification %s", fileName, contentType,
-                            fileUpload.getAccept()));
-                }
-                return false;
-            }
+
+                    return false;
+                });
+
+        if (!accepted) {
+            LOGGER.log(Level.FINE,
+                    () -> String.format("Uploaded file %s with content type %s does not match the accept specification %s",
+                            fileName, contentTypeLC, fileUpload.getAccept()));
+            return false;
         }
-        finally {
-            try {
-                Files.delete(tempFile);
-            }
-            catch (Exception ex) {
-                if (LOGGER.isLoggable(Level.WARNING)) {
-                    LOGGER.log(Level.WARNING, String.format("Could not delete temporary file %s, will try to delete on JVM exit",
-                            tempFile.toAbsolutePath()), ex);
-                    try {
-                        tempFile.toFile().deleteOnExit();
-                    }
-                    catch (Exception ex1) {
-                        if (LOGGER.isLoggable(Level.WARNING)) {
-                            LOGGER.log(Level.WARNING, String.format("Could not register temporary file %s for deletion on JVM exit",
-                                    tempFile.toAbsolutePath()), ex1);
-                        }
+
+        return true;
+    }
+
+    private static void deleteFile(Path tempFile) {
+        try {
+            Files.delete(tempFile);
+        }
+        catch (Exception ex) {
+            if (LOGGER.isLoggable(Level.WARNING)) {
+                LOGGER.log(Level.WARNING, String.format("Could not delete temporary file %s, will try to delete on JVM exit",
+                        tempFile.toAbsolutePath()), ex);
+                try {
+                    tempFile.toFile().deleteOnExit();
```

Comment:
Not exactly related to your changes, I didn't notice this as well before
But I really do think the deleteOnExit fallback should be called, even if `Level.WARNING` is not loggable